File: src/main/java/xyz/nkomarn/protocol/packet/s2c/ClientboundSetTimePacket.java
package xyz.nkomarn.protocol.packet.s2c;

import io.netty.buffer.ByteBuf;
import io.netty.buffer.Unpooled;
import org.jetbrains.annotations.NotNull;
import xyz.nkomarn.protocol.Packet;

public class ClientboundSetTimePacket extends Packet<ClientboundSetTimePacket> {

    private long time;

    public ClientboundSetTimePacket() {
    }

    public ClientboundSetTimePacket(long time) {
        this.time = time;
    }

    @Override
    public int getId() {
        return 0x04;
    }

    @Override
    public @NotNull ByteBuf encode() {
        return Unpooled.buffer()
                .writeLong(time);
    }
}

File: src/main/kotlin/kyta/composter/protocol/packet/play/ClientboundSetTimePacket.kt
package kyta.composter.protocol.packet.play

import kyta.composter.protocol.Packet
import kyta.composter.protocol.PacketSerializer
import kyta.composter.protocol.WriteBuffer

data class ClientboundSetTimePacket(val time: Long) : Packet {
    companion object : PacketSerializer<ClientboundSetTimePacket> {
        override fun serialize(packet: ClientboundSetTimePacket, buffer: WriteBuffer) {
            buffer.writeLong(packet.time)
        }
    }
}